lib/active_support/cache/redis_store.rb

```diff
@@ -249,6 +287,21 @@ def write_entry(key, entry, options)
           false
         end
 
+        def set_unless_changed(key, new_value, old_value, options)
+          with do |c|
+            c.watch(key) do
+              current_entry = read_entry(key, options)
+              if !old_value || current_entry.value == old_value
+                c.multi do
+                  write(key, new_value, options)
+                end
+              else
+                c.unwatch
```

Comment:
Won't this return a truthy value, but it should return a falsy value as the key wasn't written?